src/shared/tests.c

```diff
@@ -51,8 +51,13 @@ static void load_testdata_env(void) {
                 return;
         called = true;
 
-        assert_se(readlink_and_make_absolute("/proc/self/exe", &s) >= 0);
-        assert_se(path_extract_directory(s, &d) >= 0);
+        r = readlink_and_make_absolute("/proc/self/exe", &s);
+        if (r >= 0)
+                assert_se(path_extract_directory(s, &d) >= 0);
+        else {
+                log_debug_errno(r, "Failed to determine executable's directory, using current directory: %m");
+                assert_se(safe_getcwd(&d) >= 0);
```

Comment:
BTW, people may run a test something like `./build/test-foo`. So, maybe better to use `path_make_absolute_cwd(program_invocation_name)` and  `path_extract_directory()` ?